Tests/DatadogTests/Datadog/RUM/RUMTelemetryTests.swift

```diff
@@ -260,6 +262,60 @@ class RUMTelemetryTests: XCTestCase {
         XCTAssertEqual(events.count, 2)
     }
 
+    // MARK: - Configuration Telemetry Events
+
+    func testSendTelemetry_sendsConfigurationDelayed() throws {
+        // Given 
+        let telemetry: RUMTelemetry = .mockWith(core: core, sampler: .mockKeepAll())
+
+        // When
+        telemetry.configuration(configuration: .mockAny())
+
+        // Then immediately
+        var eventMatchers = try RUMFeature.waitAndReturnRUMEventMatchers(in: core, count: 0)
+        var events = try eventMatchers.compactMap(TelemetryConfigurationEvent.self)
+        XCTAssertEqual(events.count, 0)
+
+        // Then later
+        sleep(6)
```

Comment:
Here we add `6s` to our unit tests execution time (`12s` in total). We rather aim at keeping our tests fast (they all run under `50s` on `develop` so we're adding ~25% here). 

How about abstracting the async schedule and providing instant call here? Instead of:
```swift
DispatchQueue.main.asyncAfter(deadline: .now() + 5) { job() }
```
we could do:
```swift
class RUMTelemetry {
   /// Triggers given block asynchronously after 5s
   private let dispatchLater: (@escaping () -> Void) -> Void

   // ...
   
   func configuration(...) {
      dispatchLater { self.record(...) }
   }
}
```
And configure it async / sync respectively to the use case to avoid any wait in tests:
```swift
// async in code:
let telemetry = RUMTelemetry(
   dispatchLater: { block in DispatchQueue.main.asyncAfter(deadline: .now() + 5) { block() }
}

// sync in tests:
let telemetry = RUMTelemetry(
   dispatchLater: { block in block() }
}
```

Reply:
Yeah, I like that idea. I'll get that in shortly.